templates/kotlin/src/main/java/io/appwrite/Client.kt.twig

```diff
@@ -0,0 +1,232 @@
+            val sslSocketFactory: SSLSocketFactory = sslContext.socketFactory
+            builder.sslSocketFactory(sslSocketFactory, trustAllCerts[0] as X509TrustManager)
+            builder.hostnameVerifier(HostnameVerifier { _, _ -> true })
+
+            http = builder.build()
+        } catch (e: Exception) {
+            throw RuntimeException(e)
+        }
+
+        return this
+    }
+
+    fun setEndpoint(endPoint: String): {{ spec.title | caseUcfirst }}Client {
+        this.endPoint = endPoint
+        return this
+    }
+
+    fun addHeader(key: String, value: String): {{ spec.title | caseUcfirst }}Client {
+        headers[key] = value
+        return this
+    }
+
+    @Throws({{ spec.title | caseUcfirst }}Exception::class)
+    suspend fun call(
+        method: String, 
+        path: String, 
+        headers:  Map<String, String> = mapOf(), 
+        params: Map<String, Any?> = mapOf()
+    ): Response {
+        val requestHeaders = this.headers.toHeaders().newBuilder()
+            .addAll(headers.toHeaders())
+            .build()
+
+        val httpBuilder = (endPoint + path).toHttpUrl().newBuilder()
+
+        if ("GET" == method) {
+            params.forEach {
+                when (it.value) {
+                    null -> {
+                        return@forEach
+                    }
+                    is List<*> -> {
+                        httpBuilder.addQueryParameter(it.key + "[]", it.value.toString())
+                    }
+                    else -> {
+                        httpBuilder.addQueryParameter(it.key, it.value.toString())
+                    }
+                }
+            }
+            val request = Request.Builder()
+                .url(httpBuilder.build())
+                .headers(requestHeaders)
+                .get()
+                .build()
+
+            return awaitResponse(request)
+        }
+
+        val body = if (MultipartBody.FORM.toString() == headers["content-type"]) {
+            val builder = MultipartBody.Builder().setType(MultipartBody.FORM)
+
+            params.forEach {
+                when {
+                    it.key == "file" -> {
+                        val file = it.value as File
+                        builder.addFormDataPart(it.key, file.name, file.asRequestBody())
+                    }
+                    it.value is List<*> -> {
+                        val list = it.value as List<*>
+                        for (index in list.indices) {
+                            builder.addFormDataPart(
+                                "${it.key}[${index}]",
+                                list[index].toString()
+                            )
+                        }
+                    }
+                    else -> {
+                        builder.addFormDataPart(it.key, it.value.toString())
+                    }
+                }
+            }
+            builder.build()
+        } else {
```

Comment:
```suggestion
val body = if (MultipartBody.FORM.toString() == headers["content-type"]) {
            val builder = MultipartBody.Builder().setType(MultipartBody.FORM)
            params.forEach {
                when {
                    it.key == "file" -> {
                        val file = it.value as File
                        builder.addFormDataPart(it.key, file.name, file.asRequestBody())
                    }
                    it.value is List<*> -> {
                        val list = it.value as List<*>
                        for (index in list.indices) {
                            builder.addFormDataPart(
                                "${it.key}[]",
                                list[index].toString()
                            )
                        }
                    }
                    else -> {
                        builder.addFormDataPart(it.key, it.value.toString())
                    }
                }
            }
            builder.build()
        } else {
```

@abnegate  while reviewing the file upload functionality, I found some issues with the multipart requests. This fixes it .